test/functional/page_objects/visualize_page.js

```diff
@@ -391,27 +391,8 @@ export function VisualizePageProvider({ getService, getPageObjects }) {
       return await element.getVisibleText();
     }
 
-    // clickBucket(bucketType) 'X-Axis', 'Split Area', 'Split Chart'
-    async clickBucket(bucketName, type = 'bucket') {
-      const testSubject = type === 'bucket' ? 'bucketsAggGroup' : 'metricsAggGroup';
-      await retry.try(async () => {
-        const chartTypes = await retry.try(
-          async () => await find.allByCssSelector(`[data-test-subj="${testSubject}"] .list-group-menu-item`));
-        log.debug('found bucket types ' + chartTypes.length);
-
-        async function getChartType(chart) {
-          const chartString = await chart.getVisibleText();
-          if (chartString === bucketName) {
-            await chart.click();
-            return true;
-          }
-        }
-        const getChartTypesPromises = chartTypes.map(getChartType);
-        const clickResult = await Promise.all(getChartTypesPromises);
-        if (!clickResult.some(result => result === true)) {
-          throw new Error(`bucket ${bucketName} not found`);
-        }
-      });
+    async clickBucket(bucketName, type = 'buckets') {
+      await find.clickByCssSelector(`[group-name="${type}"] [data-test-subj="${bucketName}"]`);
```

Comment:
will this fail in case bucket to be clicked is not found ? else this will reintroduce flakiness in some tests.

Reply:
I think it should fail in that case.  I can't think of any reason we would have a test trying to click some bucket name, and it not being there would be OK.

Reply:
the problem we had before is that `clickBucket` was clicking a bucket .... but if it wouldn't find the bucket it would silently continue.

now of course we don't have any test that would be trying to click a bucket that is not there, but what happens is that due to timing issues the bucket is not there. few milliseconds later it will be.

however due to tests silently continuing here, the test will fail on the next step (which will try to click something else, which won't be there due to bucket not being clicked .... 

quite some flakiness in visualize tests was due to the implementation of clickBucket (which wouldn't wait for the bucket to click to be there and fail if it doesn't find it in certain time)

Reply:
@liza-mae and I have both tried running the Kibana tests against a Cloud instance of Kibana and had problems with the clickBucket method.   Here's the failure when I run the existing test from master against a master instance of Kibana (from about a week ago) on Cloud;

```
[09:53:16.389]        └-> "before each" hook: global before each
[09:53:16.416]        │ debg in findAllByCssSelector: [data-test-subj="bucketsAggGroup"] .list-group-menu-item
[09:53:26.327]        │ debg found bucket types 0
[09:53:26.340]        │ debg --- retry.try error: bucket X-Axis not found
[09:53:26.829]        │ debg in findAllByCssSelector: [data-test-subj="bucketsAggGroup"] .list-group-menu-item
[09:53:36.860]        │ debg found bucket types 0
[09:53:36.886]        │ debg --- retry.try failed again with the same message...
[09:53:37.362]        │ debg in findAllByCssSelector: [data-test-subj="bucketsAggGroup"] .list-group-menu-item
```

But wait!  I decided I should update my 1 week old master Cloud instance, and now the test PASSES every time without these changes.
So I guess we don't need this PR.  But I would still say that the clickBucket method as it is now is more complicated than it needs to be.  When we add data-test-subjs we're adding them to reduce our use of those cssSelectors we should take advantage of them.